src/sentry/replays/usecases/ingest.py

```diff
@@ -254,3 +224,54 @@ def process_headers(bytes_with_headers: bytes) -> tuple[RecordingSegmentHeaders,
 
 def replay_recording_segment_cache_id(project_id: int, replay_id: str, segment_id: str) -> str:
     return f"{project_id}:{replay_id}:{segment_id}"
+
+
+def store_replays_with_filestore(
+    message: RecordingIngestMessage, headers: RecordingSegmentHeaders, recording_segment: bytes
+) -> None:
+    recording_segment_file_name = f"rr:{message.replay_id}:{headers['segment_id']}"
+    with metrics.timer("replays.store_recording.store_recording.create_file"):
+        file = File.objects.create(
+            name=recording_segment_file_name,
+            type="replay.recording",
+        )
+    with metrics.timer("replays.store_recording.store_recording.put_segment_file"):
+        file.putfile(
+            BytesIO(recording_segment),
+            blob_size=settings.SENTRY_ATTACHMENT_BLOB_SIZE,
+        )
+
+    try:
+        # associate this file with an indexable replay_id via ReplayRecordingSegmentModel
+        with metrics.timer("replays.store_recording.store_recording.create_segment_row"):
+            ReplayRecordingSegmentModel.objects.create(
+                replay_id=message.replay_id,
+                project_id=message.project_id,
+                segment_id=headers["segment_id"],
+                file_id=file.id,
+                size=len(recording_segment),
+            )
+    except IntegrityError:
+        # Raised in the event of a concurrent write.  Reasonably space
+        logger.warning(
+            "Recording-segment has already been processed.",
+            extra={
+                "replay_id": message.replay_id,
+                "project_id": message.project_id,
+                "segment_id": headers["segment_id"],
+            },
+        )
+
+        # Cleanup the blob.
+        file.delete()
+
+
+def store_replays_directly(
```

Comment:
note: doing away with all postgres writes entirely for metadata and relying on the replay_id to do the lookups. this may not work for reasons, e.g. we want to do more advanced pagination and need metadata to do so, but thinking about the future of our ingestion we should try and get away from postgres. one idea would be to write to our replay_events table for each segment, either through a new row or having the sdk try to grab metadata from the recording at capture time

Reply:
another idea: bigtable HDD or some equivalent storage (or what about that new GCloud Alloy thing?